### PR TITLE
docs(README): Segregate docs to respective directories

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
### What does this PR do?

- This PR moves the command and output to respective engine directories.
- This PR adds the custom usage template to each command. With this we can provide the exact and specific information about the command.
- Removed Table of content for usage from README.md
- Added some refactors.

### Output
```
❯ kubectl openebs
openebs is a a kubectl plugin for interacting with OpenEBS storage components such as storage(pools, volumegroups), volumes, blockdevices, pvcs.
Find out more about OpenEBS on https://openebs.io/

Usage:
  kubectl openebs [command] [resource] [...names] [flags]

Available Commands:
  completion  Outputs shell completion code for the specified shell (bash or zsh)
  describe    Provide detailed information about an OpenEBS resource
  get         Provides fetching operations related to a Volume/Pool
  help        Help about any command
  version     Shows openebs kubectl plugin's version

Flags:
  -h, --help                           help for openebs
  -n, --namespace string               If present, the namespace scope for this CLI request
      --openebs-namespace string       to read the openebs namespace from user.
                                       If not provided it is determined from components.
      --cas-type                       to specify the cas-type of the engine, for engine based filtering.
                                       ex- cstor, jiva, localpv-lvm, localpv-zfs.
      --debug                          to launch the debugging mode for cstor pvcs.

Use "kubectl openebs command --help" for more information about a command.


```

### CODE REVIEW

File: cmd/completion/completion.go [hunk 1]
```diff
@@ -24,16 +24,8 @@ import (
...
-		ValidArgs: []string{"bash", "zsh"},
-		Args:      cobra.ExactValidArgs(1),
-		Short:     "Outputs shell completion code for the specified shell (bash or zsh)",
-		Long: `
-Outputs shell completion code for the specified shell (bash or zsh)
-
+const (
+	completionUsage = `
 To load completion to current bash shell,
 . <(openebs completion bash)
```
Comment (vharsh) [inline cmd/completion/completion.go]:
```suggestion
. <(kubectl openebs completion bash)
```

File: cmd/describe/describe.go [hunk 2]
```diff
@@ -17,27 +17,32 @@ limitations under the License.
...
+  kubectl openebs describe [volume|storage|pvc] [...names] [flags]
 
-# Describe a Volume:
-$ kubectl openebs describe volume pvc-abcd -n [namespace] [options]
+Describe a Volume:
+  kubectl openebs describe volume [...names] [flags]
 
-# Describe PVCs present in the same namespace:
-$ kubectl openebs describe pvc [name1] [name2] ... [nameN] -n [namespace] [options]
+Describe PVCs present in the same namespace:
```
Comment (vharsh) [inline cmd/describe/describe.go:32]:
```suggestion
Describe PVCs present in the default namespace:
```
Currently the current namespace is not picked up from the kube-context, `https://github.com/kubernetes/cli-runtime/issues/9`

File: docs/localpv-zfs/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    * #### Get `LocalPV-ZFS` volumes
+      ```bash
+      $ kubectl openebs get volumes --cas-type=zfslocalpv
+      NAMESPACE   NAME                                       STATUS   VERSION   CAPACITY   STORAGE CLASS   ATTACHED   ACCESS MODE     ATTACHED NODE
+      openebs     pvc-43fcbc72-a45a-49d5-9ec3-e383fcb91452   Ready    1.9.0     4Gi        openebs-zfspv   Bound      ReadWriteOnce   worker-sh1
+      ```
+      Note: For volumes not attached to any application, the `ATTACH NODE` would be shown as `N/A`.
+    * #### Get `LocalPV-ZFS` Pools
+      ```bash
+      $ kubectl openebs get storage --cas-type=zfslocalpv
```
Comment (vharsh) [inline docs/localpv-zfs/README.md]: The castype name has changed from `zfslocalpv` to `local-zfs`, would need to change it in multiple places.

```suggestion
      $ kubectl openebs get storage --cas-type=localpv-zfs
```

File: docs/localpv-lvm/README.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      $ kubectl openebs get volumes --cas-type=localpv-lvm
+      NAMESPACE   NAME                                       STATUS   VERSION   CAPACITY   STORAGE CLASS   ATTACHED   ACCESS MODE     ATTACHED NODE
+      openebs     pvc-04c2d4ea-f072-4e17-9e0a-db0fde0b2550   Ready    ci        1Gi        lvmpv-sc        Bound      ReadWriteOnce   worker-sh1
+      openebs     pvc-1ec1c9b7-b74e-4742-901d-2af4558d6636   Ready    ci        1Gi        openebs-lvmpv   Bound      ReadWriteOnce   worker-sh1
+      openebs     pvc-9999274f-ad01-48bc-9b21-7c51b47a870c   Ready    ci        4Gi        openebs-lvmpv   Bound      ReadWriteOnce   worker-sh1
+      ```
+      Note: For volumes not attached to any application, the `ATTACH NODE` would be shown as `N/A`.
+    * #### Get `LocalPV-LVM` VolumeGroups
+      ```bash
+      $ kubectl openebs get storage --cas-type=lvmlocalpv
```
Comment (vharsh) [inline docs/localpv-lvm/README.md]:
```suggestion
      $ kubectl openebs get storage --cas-type=localpv-lvm
```

File: docs/localpv-zfs/README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+* [LocalPV-ZFS](#localpv-zfs)
+    * [Get LocalPV-ZFS volumes](#get-localpv-zfs-volumes)
+    * [Get LocalPV-ZFS Pools](#get-localpv-zfs-pools)
+    * [Describe LocalPV-ZFS volumes](#describe-localpv-zfs-volumes)
+    * [Describe LocalPV-ZFS pools](#describe-localpv-zfs-pools)
+  
+* #### `LocalPV-ZFS`
+    * #### Get `LocalPV-ZFS` volumes
+      ```bash
+      $ kubectl openebs get volumes --cas-type=zfslocalpv
```
Comment (vharsh) [inline docs/localpv-zfs/README.md]: same

**Review by vharsh — COMMENTED** [review]
minor suggestions for casType naming, etc

File: cmd/completion/completion.go (hunk 1 above)
Comment (vharsh) [reply cmd/completion/completion.go]: These are one of the krew best practices, we need to prefix kubectl before the plugin command.

File: cmd/describe/describe.go (hunk 2 above)
Comment (Abhinandan-Purkait) [reply cmd/describe/describe.go:32]: Did not understand this change. From the word same namespace, I meant that incase the user is describing multiple PVCs for ex, `kubectl openebs describe pvc mongo-pvc cstor-pvc -n cstor`, both the PVCs should belong to the `cstor` namespace. The user cannot describe PVCs in different namespace together using `-n` flag, as the `-n` flag accepts only one value.

Comment (vharsh) [reply cmd/describe/describe.go:32]: Oh, got your context too. I had missed the `s` in the `PVCs` and I had assumed this is the example where-in namespace is not specified.

**Review by vharsh — APPROVED** [review]
lgtm

**Review by kmova — APPROVED** [review]

**Pull request merged**